src/misc/ethernet_slirp.h

```diff
@@ -0,0 +1,143 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2021  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_ETHERNET_SLIRP_H
+#define DOSBOX_ETHERNET_SLIRP_H
+
+#include "dosbox.h"
+
+#if C_SLIRP
+
+#include <map>
+#include <deque>
+#include <vector>
+#include <slirp/libslirp.h>
```

Comment:
This is incorrect. The libslirp project installs a pkg-config file that instructs third-party projects to compile using `-I${includedir}/slirp` and use headers as `#include <libslirp.h>` -- and indeed other projects using this header do it the way I just suggested.

By sheer coincidence, when installing projects on Unix to `includedir=/usr/include` that path is on the compiler's default include paths, and this bug is not noticeable.

It will however cause problems for anyone using a copy of libslirp installed to a non-default prefix, or to an overridden --includedir, or as per https://github.com/mesonbuild/wrapdb/pull/240, using meson subproject wraps.

Reply:
(Added: https://github.com/dosbox-staging/dosbox-staging/pull/1419/commits/fd054ae7f2feab72688352e4d317ea582ef93c4f)